pkg/target/http.go

```diff
@@ -411,12 +411,21 @@ func (ht *HTTPTarget) Write(messages []*models.Message) (*models.TargetWriteResu
 
 			response := response{Body: string(responseBody), Status: resp.StatusCode}
 
-			if findMatchingRule(response, ht.responseRules.Invalid) != nil {
+			// Set errors with code and body for metadata reporting
+			for _, msg := range goodMsgs {
+				msg.SetError(&models.ApiError{
+					StatusCode:   resp.Status,
+					ResponseBody: response.Body,
+					SafeMessage:  "received non-2xx response",
+				})
+			}
+
+			if matchedRule := findMatchingRule(response, ht.responseRules.Invalid); matchedRule != nil {
 				for _, msg := range goodMsgs {
 					msg.SetError(&models.ApiError{
 						StatusCode:   resp.Status,
 						ResponseBody: response.Body,
-						SafeMessage:  "matched invalid response rule",
+						SafeMessage:  matchedRule.MatchingBodyPart,
```

Comment:
So we use body part for invalid rules, now can we also do the same for setup rules?